lib/workos/directory_user.rb

```diff
@@ -47,6 +47,15 @@ def to_json(*)
       }
     end
 
+    def primary_email
+      return nil if emails.empty?
+
+      primary_email = emails.find { |email| email[:primary] }
+      return primary_email[:value] if primary_email
```

Comment:
❤️ :shipit:

Reply:
We can also maybe shorten this up a bit I think, if we're interested. A couple notes:

1. Ruby will implicity return a nil so need to specify.
2. We don't need to check if emails is empty because an empty array will return nil if nothing is found with `find`

I believe we could write:

```rb
def primary_email(emails)
  (emails || []).find { |email| email[:primary] }
end
```

Some quick tests:

```
irb(main):002:0> emails = [{ email: 'foo@foo-corp.com', primary: true }, { email: 'foo@gmail.com' }]
=> [{:email=>"foo@foo-corp.com", :primary=>true}, {:email=>"foo@gmail.com"}]
irb(main):003:0> empty_emails = []
=> []
irb(main):004:0> nil_emails = nil
=> nil
irb(main):005:0> def primary_email(emails); (emails || []).find { |email| email[:primary] }; end
=> :primary_email
irb(main):006:0> primary_email(emails)
=> {:email=>"foo@foo-corp.com", :primary=>true}
irb(main):007:0> primary_email(empty_emails)
=> nil
irb(main):008:0> primary_email(nil_emails)
=> nil
```

Reply:
(We also don't need the `|| []` if we're sure emails is never `nil`)

Reply:
I think there's some question as to whether this line is doing quite what we want:

```
irb(main):006:0> primary_email(emails)
=> {:email=>"foo@foo-corp.com", :primary=>true}
```

If we're ok with returning the entire email Hash here, then it works - if we want to return just the email address, then we end up needing to extract it from the hash, which gets us back to being kinda messy (and how we'd landed on the solution ☝️ in the first place - https://github.com/workos/workos-ruby/pull/161#discussion_r894883923). But if we're ok returning the entire Hash, then I'm definitely 👍 for this shorter/cleaner version.

My (entirely personal) preference is to keep `return nil if emails.empty?` at the top of the method - just makes it [visually] more clear what will happen if we have an empty array. I suppose we actually need to do `emails.blank?` though if we do need to worry about `email` being nil (`nil.empty?` is a 💥 😭).

Reply:
Thanks! I added those changes 👍 and thanks for the Ruby knowledge! I think we verify here that emails is an array: https://github.com/workos/workos-ruby/blob/main/lib/workos/types/directory_user_struct.rb#L11
but I just kept the check.

Reply:
Yeah as a follow up - we would verify that the emails attribute is an array before creating the directory user object, so it shouldn't be `nil`. I can revert back to Alex's suggestion. Whatever works best 👍

Reply:
Meh 🤷‍♂️, having the `|| []` in there doesn't do any harm, and makes us more future-proof, in case [for some unknown reason!] we decided to change that logic!

Reply:
Let me know if the current version looks good, or if we want to iterate further. Thanks!